sql/src/main/java/org/apache/druid/sql/calcite/aggregation/builtin/BuiltinApproxCountDistinctSqlAggregator.java

```diff
@@ -151,12 +165,22 @@ private static class BuiltinApproxCountDistinctSqlAggFunction extends SqlAggFunc
           SqlKind.OTHER_FUNCTION,
           ReturnTypes.explicit(SqlTypeName.BIGINT),
           InferTypes.VARCHAR_1024,
-          OperandTypes.ANY,
+          OperandTypes.or(
+              OperandTypes.STRING,
+              OperandTypes.NUMERIC,
+              RowSignatures.complexTypeChecker(HyperUniquesAggregatorFactory.TYPE)
+          ),
           SqlFunctionCategory.STRING,
           false,
           false,
           Optionality.FORBIDDEN
       );
     }
   }
+
+  private boolean validateInputType(ColumnType columnType)
```

Comment:
I think this method name is a bit unfortunate; 

the method name suggests that it will deiced if an input type is valid; however it only accepts some complex types - I would expect a `true` from this method for simple types like `LONG` .

it should be either renamed; or reorganized

Reply:
Renamed.